### PR TITLE
Added toversion to all testplaybooks under nonCircleTest

### PR DESCRIPTION
**Pull request by barchen1**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/29088

## Description
Added `toversion:4.5.9` to all testplaybooks under NonCircleTests and update the old_content_branch script.

### CODE REVIEW

File: TestPlaybooks/NonCircleTests/Deprecated/playbook-Indicator_Feed_-_Test.yml [hunk 1]
```diff
@@ -1,486 +0,0 @@
-id: Indicator Feed - Test
```
Comment (Itay4) [inline TestPlaybooks/NonCircleTests/Deprecated/playbook-Indicator_Feed_-_Test.yml:1]: why did you delete this one ?

File: TestPlaybooks/NonCircleTests/Deprecated/playbook-McAfee-ePOTest.yml [hunk 2]
```diff
@@ -1,472 +0,0 @@
-id: Test Playbook McAfee ePO
```
Comment (Itay4) [inline TestPlaybooks/NonCircleTests/Deprecated/playbook-McAfee-ePOTest.yml:1]: why did you delete this one?

File: Utils/old_content_branch.py [hunk 3]
```diff
@@ -226,6 +226,11 @@ def edit_reputations_json(new_to_version):
         ujson.dump(rep_content, f, indent=4, encode_html_chars=True)
 
 
+def edit_non_circle_tests(new_to_version):
```
Comment (Itay4) [inline Utils/old_content_branch.py:229]: do we have unit test for this ?

**Comment by ronykoz** [comment]
@barchen1 what's the status here?

**Comment by barchen1** [comment]
> @barchen1 what's the status here?
Waiting for this PR to merge https://github.com/demisto/content/pull/9320/files and adding a unitest for edit_non_circle_tests func

File: Utils/old_content_branch.py [hunk 4]
```diff
@@ -223,7 +223,13 @@ def edit_reputations_json(new_to_version):
             reputation['toVersion'] = new_to_version
 
     with open(rep_json_path, 'w') as f:
-        ujson.dump(rep_content, f, indent=4, encode_html_chars=True)
+        ujson.dump(rep_content, f, indent=4, encode_html_chars=True, escape_forward_slashes=False,
+                   ensure_ascii=False)
+
+
+def edit_non_circle_tests(new_to_version):
```
Comment (Itay4) [inline Utils/old_content_branch.py:230]: please add type hints and docstring

File: Utils/old_content_branch.py [hunk 5]
```diff
@@ -223,7 +223,13 @@ def edit_reputations_json(new_to_version):
...
+    edit_playbooks_directory(new_to_version, 'TestPlaybooks')
```
Comment (Itay4) [inline Utils/old_content_branch.py:231]: lets set those strings to be consts

File: Utils/tests/old_content_branch_test.py [hunk 6]
```diff
@@ -64,5 +70,24 @@ def test_handle_yml__should_delete():
...
+    - A path to a tpb that should stay and its toversion field should change.
+
+    When
+    - Running edit_playbooks_directory on it.
+
+    Then
+    - Tpb should be updated with a new toversion field
+    """
+    edit_playbooks_directory(TEST_VERSION, 'Utils/tests/test_data_old_content')
+    with open(TEMP_TPB, 'r') as yml_file:
```
Comment (Itay4) [inline Utils/tests/old_content_branch_test.py:91]: lets use pytest fixture `tmp_path` instead

**Review by Itay4 — CHANGES_REQUESTED** [review]

**Comment by Itay4** [comment]
note that this unit test isnt actually ran in the build, please add it to `Infrastructure testing`

File: Utils/old_content_branch.py (hunk 5 above)
Comment (barchen1) [reply Utils/old_content_branch.py]: done

File: Utils/old_content_branch.py (hunk 4 above)
Comment (barchen1) [reply Utils/old_content_branch.py]: done

File: Utils/tests/old_content_branch_test.py (hunk 6 above)
Comment (barchen1) [reply Utils/tests/old_content_branch_test.py]: why should we use tmp_path ?
its not really a temp path, ill change the name

File: Utils/old_content_branch.py [hunk 7]
```diff
@@ -9,27 +11,34 @@
...
+ryaml.preserve_quotes = True  # type: ignore
+# make sure long lines will not break (relevant for code section)
+ryaml.width = 50000  # type: ignore
+
+NON_CIRCLE_TEST_PLAYBOOKS_DIRECTORY = 'TestPlaybooks'
+DEPRECATED_NON_CIRCLE_TESTS_DIRECTORY = os.path.join('TestPlaybooks', 'NonCircleTests', 'Deprecated')
 
 
-def should_keep_yml_file(yml_content, new_to_version):
+def should_keep_yml_file(yml_content: Dict, new_to_version: str):
```
Comment (Itay4) [inline Utils/old_content_branch.py:22]:
```suggestion
def should_keep_yml_file(yml_content: Dict, new_to_version: str) -> bool:
```

File: Utils/old_content_branch.py [hunk 8]
```diff
@@ -9,27 +11,34 @@
...
+    """Check if yml file should stay in the feature branch"""
     if parse_version(yml_content.get('toversion', '99.99.99')) < parse_version(new_to_version) or \
             parse_version(yml_content.get('fromversion', '0.0.0')) >= parse_version(new_to_version):
         return False
 
     return True
 
 
-def should_keep_json_file(json_content, new_to_version):
+def should_keep_json_file(json_content: Dict, new_to_version: str):
```
Comment (Itay4) [inline Utils/old_content_branch.py:31]:
```suggestion
def should_keep_json_file(json_content: Dict, new_to_version: str) -> bool:
```

File: Utils/tests/old_content_branch_test.py (hunk 6 above)
Comment (Itay4) [reply Utils/tests/old_content_branch_test.py]: because we shouldnt modify actual files, it will require us to undo the changes every time we want to run the test
we should modify temp files that pytest caches over tests

**Review by Itay4 — APPROVED** [review]
